app/src/main/proto/vss.proto

```diff
@@ -0,0 +1,182 @@
+    It is up to clients to use single or multiple stores for their use-case.
+    This can be used for client-isolation/ rate-limiting / throttling on the server-side.
+    Authorization and billing can also be performed at the storeId level.
+ */
+  string storeId = 1;
+
+  /*
+    Key for which the value is to be fetched.
+
+    Consistency Guarantee:
+    Get/Read operations against a key are consistent reads and will reflect all previous writes,
+    since Put/Write provides read-after-write and read-after-update consistency guarantees.
+
+    Read Isolation:
+    Get/Read operations against a key are ensured to have read-committed isolation.
+    Ref: https://en.wikipedia.org/wiki/Isolation_(database_systems)#Read_committed
+   */
+  string key = 2;
+}
+
+
+message GetObjectResponse {
+  /*
+    Fetched value and version along with the corresponding key in the request.
+   */
+  KeyValue value = 2;
+}
+
+
+message PutObjectRequest {
+
+  /*
+    StoreId: All APIs operate within a single storeId.
+    StoreId is a keyspace identifier.
+    Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+    It is up to clients to use single or multiple stores for their use-case.
+    This can be used for client-isolation/ rate-limiting / throttling on the server-side.
+    Authorization and billing can also be performed at the storeId level.
+ */
+  string storeId = 1;
+  /*
+    If present, the write will only succeed if the current server-side globalVersion against
+    a storeId is same as in the request.
+    Clients are expected to store (client-side) the global version against storeId.
+    And the request should contain their client-side value of globalVersion.
+
+    For the first write of the store, global version should be '0'. If the write succeeds, clients
+    must increment their global version (client-side) by 1.
+    The server increments globalVersion (server-side) for every successful write, hence this
+    client-side increment is required to ensure matching versions. This updated global version
+    should be used in subsequent PutObjectRequests for the store.
+
+    Requests with conflicting version will fail with `ConflictException` as ErrorCode.
+   */
+  optional int64 globalVersion = 2;
+
+  /*
+    Items to be written as a result of this PutObjectRequest.
+
+    In an item, each key is supplied with its corresponding value and version.
+    Clients can choose to encrypt the keys client-side in order to obfuscate user's usage patterns.
+    If write is successful, previous value corresponding to the key will be overwritten.
+
+    Multiple items in transactionItems of a single PutObjectRequest are written in
+    a database-transaction in all-or-nothing fashion.
+    Items in single PutObjectRequest must have distinct keys.
+
+    Write will only succeed if current DB version against key is same as in request.
+    Client is expected to store version against every key.
```

Comment:
s/Client is/Clients are
s/version/versions

Reply:
Since only single version against a key needs to be stored. 
I think current version is correct ? 
Or 
"Clients are expected to store a version against every key."